### PR TITLE
Corrected errors in deployment scripts, added multiple fixes from audits to contracts

### PR DESCRIPTION
**Pull request by jrchatruc**
# Corrected errors in deployment scripts, added multiple fixes from audits to contracts

This PR corrects erros in the deployment scripts. Additionally, it addersses the following issues with the contracts:

- TOTAL_SUPPLY constant is removed since it's redundant
- Solidity version was outdated
- Pragma versions where not fixed
- Missing docstrings were added
- TokenClaimed  event parameters were documented

### CODE REVIEW

**Review by juanbono — APPROVED** [review]

**Review by ilitteri — APPROVED** [review]

File: claim_contracts/script/UpgradeToAlignedTokenV3.s.sol [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// import "../src/TestToken.sol";
+import "../src/ExampleAlignedTokenV3.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {ERC1967Utils} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Utils.sol";
+import "forge-std/Script.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {Utils} from "./Utils.sol";
+
+/// @notice This script upgrades the ClaimableAirdrop contract to ClaimableAirdropV2.
```
Comment (MegaRedHand) [inline claim_contracts/script/UpgradeToAlignedTokenV3.s.sol:13]: This comment is wrong

File: claim_contracts/src/AlignedToken.sol [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
 
 contract AlignedToken is
     Initializable,
     ERC20Upgradeable,
-    EIP712Upgradeable,
```
Comment (MegaRedHand) [inline claim_contracts/src/AlignedToken.sol:12]: Is it OK to remove this?

**Review by MegaRedHand — APPROVED** [review]

File: claim_contracts/src/AlignedToken.sol (hunk 2 above)
Comment (ilitteri) [reply claim_contracts/src/AlignedToken.sol:12]: Yep! It comes for free with ERC20PermitUpgradeable.

**Comment by ilitteri** [comment]
Fixes OpenZeppelin issues: 
- L-01
- L-04
- L-05
- L-06
- N-012

**Comment by ilitteri** [comment]
Fixes Least Authority Suggestion 2

**Pull request merged**